src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Completion/RazorCompletionListProvider.cs

```diff
@@ -77,7 +76,8 @@ internal class RazorCompletionListProvider(
         var completionCapability = clientCapabilities?.TextDocument?.Completion as VSInternalCompletionSetting;
 
         // The completion list is cached and can be retrieved via this result id to enable the resolve completion functionality.
-        var razorResolveContext = new RazorCompletionResolveContext(documentContext.FilePath, razorCompletionItems);
+        var filePath = codeDocument.Source.FilePath.AssumeNotNull();
```

Comment:
👀 This means that any `RazorCodeDocument` passed in must have a `RazorSourceDocument` with an accurate file path. Both `ProjectSnapshotManager` and co-hosting guarantee this fact, but it required some extra tweaking in tests.